mods/creative/inventory.lua

```diff
@@ -59,22 +59,34 @@ function creative.init_creative_inventory(player)
 	return player_inventory[player_name]
 end
 
+local function match(s, filter)
+	if filter == "" then return 0 end
+	if s:lower():find(filter, 1, true) then
+		return #s - #filter
+	end
+	return nil
+end
+
 function creative.update_creative_inventory(player_name, tab_content)
-	local creative_list = {}
 	local inv = player_inventory[player_name] or
 			creative.init_creative_inventory(minetest.get_player_by_name(player_name))
 	local player_inv = minetest.get_inventory({type = "detached", name = "creative_" .. player_name})
 
 	local items = inventory_cache[tab_content] or init_creative_cache(tab_content)
 
+	local creative_list = {}
+	local order = {}
 	for name, def in pairs(items) do
-		if def.name:find(inv.filter, 1, true) or
-				def.description:lower():find(inv.filter, 1, true) then
+		local m = match(def.name, inv.filter) or match(def.description, inv.filter)
```

Comment:
Preferring the weighting of the description probably makes more sense.
```suggestion
		local m = match(def.description, inv.filter) or match(def.name, inv.filter)
```
Example:
search for "sand":
current:
default:sand (Sand): 8
foo:bar (Sand Tool): 5 \<- first
as suggestion:
default:sand (Sand): 0 \<- first
foo:st (Sand Tool): 5